api/v1beta1/machine_types.go

```diff
@@ -56,6 +56,22 @@ const (
 	// This annotation can be set on BootstrapConfig or Machine objects. The value set on the Machine object takes precedence.
 	// This annotation can only be used on Control Plane Machines.
 	MachineCertificatesExpiryDateAnnotation = "machine.cluster.x-k8s.io/certificates-expiry"
+
+	// NodeRoleLabelPrefix is one of the CAPI managed prefix Node labels.
+	NodeRoleLabelPrefix = "node-role.kubernetes.io/"
+	// NodeRestrictionLabelPrefix is one of the CAPI managed prefix Node labels.
+	NodeRestrictionLabelPrefix = "node-restriction.kubernetes.io/"
+	// NodeCAPILabelPrefix is one of the CAPI managed prefix Node labels.
+	NodeCAPILabelPrefix = "node.cluster.x-k8s.io/"
```

Comment:
```suggestion
	// ManagedNodeLabelPrefix is the prefix for labels that are managed by Cluster API and synced to Kubernetes Node object at every reconciliation.
	ManagedNodeLabelPrefix = "node.cluster.x-k8s.io/"
```

This should be the only (continuously) managed prefix

Reply:
Fabrizio, Alberto and I discussed this this afternoon. The plan for now is to take a look at Server-Side Apply. By using SSA we could manage all 3 prefixes without overwriting additional labels set by users / other controllers.

Reply:
That make sense, thank you!